web/packages/teleport/src/services/databases/makeDatabase.ts

```diff
@@ -16,19 +16,21 @@ limitations under the License.
 
 import { formatDatabaseInfo } from 'shared/services/databases';
 
-import { Database, DatabaseService } from './types';
+import { Aws, Database, DatabaseService } from './types';
 
 export function makeDatabase(json: any): Database {
   const { name, desc, protocol, type, aws } = json;
 
   const labels = json.labels || [];
 
   // Only setting RDS fields for now.
-  let rds;
+  let awsRds: Aws;
```

Comment:
i just mimicked the type that was returned to me,  we get this [AWS](https://github.com/gravitational/teleport/blob/5ba028bc5843db6506f4d41919221286ce60bb40/api/proto/teleport/legacy/types/types.proto#L388) type from the back (which has a rds field in it)

i think it makes sense to leave it as is for now, a non-empty `aws` field lets us know that this db is hosted by aws, and rds field describes what type of rds.

Reply:
Got it. I'm not actually sure about this:

```proto
  // RDS contains RDS specific metadata.
  RDS RDS = 3 [
    (gogoproto.nullable) = false,
    (gogoproto.jsontag) = "rds,omitempty"
  ];
```

If we have `nullable = false` but also `omitempty`, will it ever come back empty?

Reply:
yes, for the web api, it will return the field as `undefined` b/c we check if it's aws hosted before setting the field:
https://github.com/gravitational/teleport/blob/b3e83faead6527fb25c4d5b7f04da711757c6606/lib/web/ui/server.go#L281

Reply:
I see that that answers that `aws` might be `undefined` when being sent back via the API, my question is whether `aws.rds` itself can ever be empty. It seems like it can't (even though we have `omitempty`), I think it will always be part of the `aws` object if that's returned, it will just have go's default values.

Ultimately what I'm asking is whether we should remove the optionality of `rds` from the `Aws` object, making it:

```typescript
export type Aws = {
  rds: Pick<AwsRdsDatabase, 'resourceId' | 'region' | 'subnets'>;
};
```

I don't see anywhere that it's being treated as `undefined`.

Reply:
oh i get what you're saying now, yes you're completely right. edited: https://github.com/gravitational/teleport/pull/27924/commits/874231b0cb9086c76739dced5dfdcd88a04e02ec